pandas/core/categorical.py

```diff
@@ -2066,3 +2066,48 @@ def _factorize_from_iterables(iterables):
         # For consistency, it should return a list of 2 lists.
         return [[], []]
     return map(list, lzip(*[_factorize_from_iterable(it) for it in iterables]))
+
+
+class CategoricalType(CategoricalDtype):
+    """
```

Comment:
this should be in with the rest of the dtypes

to be honest i wouldn't create this; just add it optionally into the existing